src/SonarQube.Client.Tests/Services/SonarQubeClientTests.cs

```diff
@@ -509,5 +511,119 @@ private void Method_WhenExceptionThrown_PropagateIt<T>(Func<SonarQubeClient, Can
+            };
+
+            var expected = new NotificationsResponse[]
+            {
+                new NotificationsResponse
+                {
+                    Category = "QUALITY_GATE",
+                    Message =  "Quality Gate of project 'test' is now Red (was Green)",
+                    Link = new Uri("http://localhost:9000/dashboard?id=test"),
+                    Date = new DateTimeOffset(2017, 9, 14, 10, 55, 19, 0, TimeSpan.FromHours(2)),
+                    Project = "test"
+                }
+            };
+
+            // Act
+            var result = await client
+                .GetNotificationEventsAsync(request, CancellationToken.None);
+
+            // Assert
+            result.IsSuccess.Should().BeTrue();
+            AssertEqual(expected, result.Value);
+        }
+
+        [TestMethod]
+        public async Task GetNotificationEventsAsync_NeverReturnsNull()
+        {
+            foreach (var statusCode in Enum.GetValues(typeof(HttpStatusCode)).Cast<HttpStatusCode>())
+            {
+                var result = await GetNotificationEventsAsync_WithHttpStatusCode(statusCode);
+                result.Value.Should().BeEmpty($"failed for {statusCode}");
+            }
+        }
+
+        private static async Task<Result<NotificationsResponse[]>> GetNotificationEventsAsync_WithHttpStatusCode(
+            HttpStatusCode returnedStatusCode)
+        {
+            // Arrange
+            var responseContent =
+                returnedStatusCode == HttpStatusCode.OK ? "{\"events\":[]}" : null;
+
+            var client = SetupClientWithHttpResponse(returnedStatusCode, responseContent);
+            var request = new NotificationsRequest { ProjectKey = "project", EventsSince = DateTimeOffset.Now };
+
+            // Act
+            return await client.GetNotificationEventsAsync(request, CancellationToken.None);
+        }
+
+        private static SonarQubeClient SetupClientWithHttpResponse(HttpStatusCode code, string responseText)
+        {
+            var response = new HttpResponseMessage(code);
+            if (responseText != null)
+            {
+                response.Content = new StringContent(responseText);
+            }
+
+            var httpHandler = new Mock<HttpMessageHandler>();
+            httpHandler.Protected()
+                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(),
+                    ItExpr.IsAny<CancellationToken>())
+                .Returns(Task<HttpResponseMessage>.Factory.StartNew(() => response));
+
+            var connection = new ConnectionRequest { ServerUri = new Uri("http://mysq.com/") };
+            return new SonarQubeClient(connection, httpHandler.Object, TimeSpan.FromSeconds(10));
+        }
+
+        private void AssertEqual(NotificationsResponse x, NotificationsResponse y, int itemIndex)
+        {
+            Assert.AreEqual(x.Category, y.Category, string.Format("Category, item {0}", itemIndex));
```

Comment:
Why use `Assert.AreEqual` and not FluentAssertions?

Reply:
The error message from fluent assertions was horribly unhelpful.